src/utils/backwardCompatability.js

```diff
@@ -0,0 +1,124 @@
+    }
+
+    >= 2.34
+    "reportingParams": {
+        "grandParentOrganisationUnit": false,
+        "reportingPeriod": false,
+        "organisationUnit": false,
+        "parentOrganisationUnit": false
+    }
+
+3. From >= 2.34 onward we have to deal with some inconsistencies in the property names:
+    - `report.reportParams`
+    - `report.visualization.reportingParams`
+
+The functions in this module are there to deal with these discrepancies. To avoid too many changes to the app state model, the following names have been settled on:
+1. `reportTables` in favour of `visualizations`
+2. `reportParams` in favour of `reportingParams`
+3. don't prefix reportParams properties with `param`, 
+   i.e. `reportingPeriod` in favour of `paramReportingPeriod`
+
+
+When we drop support for version v2.33 these helpers can be removed and replaced
+with much more simple and sane logic
+*/
+
+const isBelowVersion34 = () => getSystemMinorVersion() < 34
+
+const getReportTableNameForSystemVersion = () =>
+    isBelowVersion34() ? 'reportTable' : 'visualization'
+
+const getReportParamsFieldName = () =>
+    isBelowVersion34() ? 'reportParams' : 'reportingParams'
+
+export const getReportTablesResourceNameForSystemVersion = () =>
+    `${getReportTableNameForSystemVersion()}s`
+
+/**
+ * Required fields for displaying the standard reports
+ */
+export const getStandardReportsFieldsForSystemVersion = () => {
+    const reportTableFieldName = getReportTableNameForSystemVersion()
+    const reportParamsFieldName = getReportParamsFieldName()
+
+    return [
+        'displayName',
+        'type',
+        'id',
+        `${reportTableFieldName}[id,displayName,${reportParamsFieldName},relativePeriods]`,
+        'reportParams',
+        'relativePeriods',
+        'access',
+    ]
+}
+
+const getReportParamsPropertiesForSystemVersion = reportParams => {
+    if (!isBelowVersion34() || !reportParams) {
+        return reportParams
+    }
+
+    return {
+        reportingPeriod: reportParams.paramReportingPeriod,
+        grandParentOrganisationUnit:
+            reportParams.paramGrandParentOrganisationUnit,
+        organisationUnit: reportParams.paramOrganisationUnit,
+        parentOrganisationUnit: reportParams.paramParentOrganisationUnit,
+    }
+}
+
+export const formatStandardReportResponseForSystemVersion = reportModel => {
+    // handle both d2 model instances and plain objects
+    const reportJson = reportModel.toJSON ? reportModel.toJSON() : reportModel
+    // <2.34 reportModel.reportTable, >=2.34 reportmodel.visualization
+    const reportTable = reportModel[getReportTableNameForSystemVersion()]
+    // <2.34 reportModel.reportTable.reportParams, >=2.34 reportmodel.visualization.reportingParams
+    const reportTableReportParams =
+        reportTable && reportTable[getReportParamsFieldName()]
+
+    return {
+        ...reportJson,
+        reportParams: getReportParamsPropertiesForSystemVersion(
+            reportJson.reportParams
```

Comment:
Does this mean that the d2 model uses the `< 2.34` names? This is so confusing..